### PR TITLE
ENG-4874 and ENG-4875 Update and create contacts

### PR DESCRIPTION
**Pull request by MEM2677**
@avdev4j here is the PR to ENG-4879, I have also added some silly tests for the mapper

### CODE REVIEW

File: microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/config/SecurityConfiguration.java [hunk 1]
```diff
@@ -14,9 +14,8 @@ public class SecurityConfiguration {
 
     @Bean
     public SecurityFilterChain configure(HttpSecurity http) throws Exception {
-        return http.csrf()
-                .disable()
-                .securityMatcher("/**")
+
```
Comment (avdev4j) [inline microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/config/SecurityConfiguration.java:17]: I think we should keep as we will have the same issue with the UI?

File: microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/controller/ContactController.java [hunk 2]
```diff
@@ -56,4 +61,12 @@ public ResponseEntity<String> deleteContact(@PathVariable Long id) {
 
         return ResponseEntity.noContent().build();
     }
+
+
+  @PostMapping("/contact")
+    public ResponseEntity<ContactRecord> createContact(@RequestBody ContactRecord contact) {
+        log.debug("REST request to create a NEW contact: {}", contact );
+        ContactRecord created = contactService.save(contact);
+        return new ResponseEntity<>(created, HttpStatus.CREATED);
```
Comment (avdev4j) [inline microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/controller/ContactController.java:70]:
```suggestion
        return ResponseEntity.created(new URI("/api/contacts/" + created.id())).build();```

File: microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/config/SecurityConfiguration.java (hunk 1 above)
Comment (MEM2677) [reply microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/config/SecurityConfiguration.java:17]: Done

File: microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/controller/ContactController.java (hunk 2 above)
Comment (MEM2677) [reply microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/controller/ContactController.java]: Done

File: microservices/springboot-agenda/src/test/java/com/entando/springbootagenda/service/mapper/ContactMapperIT.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package com.entando.springbootagenda.service.mapper;
```
[the remark this answers is not in the document]
Comment (MEM2677) [reply microservices/springboot-agenda/src/test/java/com/entando/springbootagenda/service/mapper/ContactMapperIT.java]: Done

File: microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/service/ContactServiceImpl.java [hunk 4]
```diff
@@ -34,4 +39,11 @@ public void delete(Long id) {
         contactRepository.findOneById(id).ifPresent(contactRepository::delete);
     }
 
+    @Override
+    public ContactRecord save(ContactRecord contact) {
+        ContactEntity entity = contactMapper.toEntity(contact);
+        ContactEntity saved = contactRepository.save(entity);
+        return new ContactRecord(saved);
```
[the remark this answers is not in the document]
Comment (MEM2677) [reply microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/service/ContactServiceImpl.java:46]: Usually is a problem of upper layers to invoke the service with correct data, as developer I usually prefer the service throwing an exception when invoked with a null object... However I leave it up to you  

I could modify the code this way

`    public ContactRecord saveOpt(Optional<ContactRecord> contact) {
        return contact
                .map(c -> contactRepository.save(contactMapper.toEntity(c)))
                .map(ContactRecord::new)
                .orElse(null);
    }`

...and change the controller accordingly

[the remark this answers is not in the document]
Comment (avdev4j) [reply microservices/springboot-agenda/src/main/java/com/entando/springbootagenda/service/ContactServiceImpl.java:46]: no, finally I changed my mind. Optional is not a good idea here :D

**Pull request closed without merging**